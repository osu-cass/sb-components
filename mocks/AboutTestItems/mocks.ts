import {
  Resource,
  AboutItemModel,
  AboutTestItemsModel,
  InteractionTypeModel,
  AboutTestItemsParams,
  AboutTestSearchParams
} from "@src/index";
import { action } from "@storybook/addon-actions";
import { match } from "react-router";
import {
  aboutItemMockModel,
  mockPromiseLoading,
  mockPromiseReject
} from "@mocks/index";

const interactionTypes: InteractionTypeModel[] = [
  {
    code: "EBSR",
    label: "Evidence-Based Selected Response",
    description:
      "<p>This item presents students with a two-part question, parts A and B. \n              Students select a correct response from four options in Part A and then identify textual support for that response in Part B.</p>",
    order: 1
  },
  {
    code: "EQ",
    label: "Equation",
    description:
      "<p>The Equation item type has one or more text boxes for a response area and a key pad containing mathematical characters. \n              The student enters an equation or numerical answer into the text box using the key pad or keyboard.</p>",
    order: 2
  }
];

export const defaultAboutTestItemsModel: AboutTestItemsModel = {
  interactionTypes,
  itemUrl: "  ",
  aboutThisItemViewModel: aboutItemMockModel,
  selectedInteractionTypeCode: "EQ"
};

async function mockAboutClient(
  params?: AboutTestSearchParams
): Promise<AboutTestItemsModel> {
  const aboutItems = { ...defaultAboutTestItemsModel };
  if (params) {
    aboutItems.selectedInteractionTypeCode = params.interactionTypeCode;
  }

  return aboutItems;
}

export const mockAboutClientSuccess = mockAboutClient;

export const mockAboutRejectClient = (params?: AboutTestSearchParams) =>
  mockPromiseReject<AboutTestItemsModel>("Canceled");

export const mockAboutLoading = (params?: AboutTestSearchParams) =>
  mockPromiseLoading<AboutTestItemsModel>();

export const aboutTestPath = "/:itemType?";

export const aboutTestMatch: match<AboutTestItemsParams> = {
  params: { itemType: "EQ" },
  isExact: true,
  path: "aboutTestPath",
  url: "/"
};

export const aboutTestBadItem: match<AboutTestItemsParams> = {
  ...aboutTestMatch,
  params: { itemType: "somebadtype" }
};
