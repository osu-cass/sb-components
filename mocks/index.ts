//
// About Item
//
export {
  rubricsWithSamples,
  mockRubrics,
  rubricsEsn,
  aboutItemRevisionMockModel,
  aboutItemMockModel,
  aboutItemMockNoRubric,
  allRubrics
} from "./AboutItem/mocks";

//
// About Test Items
//
export {
  defaultAboutTestItemsModel,
  aboutTestBadItem
} from "./AboutTestItems/mocks";

//
// Accessibility
//
export {
  mockAccResourceGroups,
  allAccessibilityResourceGroups
} from "./Accessibility/mocks";

//
// Item Bank
//
export {
  mockBankAboutItemClient,
  mockBankAccessibilityClient,
  mockBankRevisionsClient,
  mockBankSectionsClient,
  itemRevisionMocks,
  namespaceMocks,
  sectionMocks
} from "./ItemBank/mocks";

//
// Item Card
//
export {
  completeItemCard,
  itemCardList,
  sortableItemCards
} from "./ItemCard/mocks";

//
// Revisions
//
export { mockRevisions } from "./Revisions/RevisionMocks";

//
// Promise
//

export { mockPromise, mockPromiseLoading, mockPromiseReject } from "./promise";
