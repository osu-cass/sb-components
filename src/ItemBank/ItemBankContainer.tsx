import * as React from "react";
import {
  getResourceContent,
  AccResourceGroupModel,
  AboutItemRevisionModel,
  Resource,
  Subscription,
  ItemRevisionModel,
  itemRevisionKey,
  NamespaceModel,
  SectionModel,
  AccessibilityRevisionModel,
  validItemRevisionModel,
  RevisionModel,
  ItemBankViewer,
  ItemBankEntry,
  getNextItemBank,
  getPreviousItemBank,
  toiSAAP
} from "@src/index";
import { itemsAreEqual } from "@src/ItemBank/ItemBankModels";

export interface ItemBankContainerProps {
  accessibilityClient: (
    acc: AccessibilityRevisionModel
  ) => Promise<AccResourceGroupModel[]>;
  aboutItemRevisionClient: (
    item: ItemRevisionModel
  ) => Promise<AboutItemRevisionModel>;
  revisionsClient: (item: ItemRevisionModel) => Promise<RevisionModel[]>;
  namespacesClient: () => Promise<NamespaceModel[]>;
  sectionsClient: () => Promise<SectionModel[]>;
  itemViewUrl?: string;
  items?: ItemRevisionModel[];
  setUrl: (item: ItemRevisionModel) => void;
  resetUrl: () => void;
}

export interface ItemBankContainerState {
  aboutItemRevisionModel: Resource<AboutItemRevisionModel>;
  accResourceGroups: Resource<AccResourceGroupModel[]>;
  currentItem?: ItemRevisionModel;
  csvText: string;
  items: ItemRevisionModel[];
  namespaces: Resource<NamespaceModel[]>;
  sections: Resource<SectionModel[]>;
  revisions: Resource<RevisionModel[]>;
  nextItem?: ItemRevisionModel;
  previousItem?: ItemRevisionModel;
  hasError: boolean;
}

/**
 * Item Bank Entry Page for displaying items, revisions, and accessibility
 * @export
 * @class ItemBankContainer
 * @extends {React.Component<ItemBankContainerProps, ItemBankContainerState>}
 */
export class ItemBankContainer extends React.Component<
  ItemBankContainerProps,
  ItemBankContainerState
> {
  private subscription = new Subscription();

  constructor(props: ItemBankContainerProps) {
    super(props);
    const items = props.items || [{}];
    const currentItem = validItemRevisionModel(items[0]) ? items[0] : undefined;
    this.state = {
      currentItem,
      items,
      csvText: "",
      aboutItemRevisionModel: { kind: "loading" },
      accResourceGroups: { kind: "loading" },
      namespaces: { kind: "loading" },
      sections: { kind: "loading" },
      revisions: { kind: "loading" },
      hasError: false
    };
  }

  componentDidMount() {
    this.fetchNamespaces().catch(e => this.onError(e));
    this.fetchSections().catch(e => this.onError(e));
    this.handleChangeViewItem();
    this.handleChangeRevision();
  }

  componentWillUnmount() {
    this.subscription.cancelAll();
  }

  async fetchAboutItemRevisionModel(
    item: ItemRevisionModel
  ): Promise<AboutItemRevisionModel> {
    const prom = this.props.aboutItemRevisionClient(item);
    const promiseWrapper = this.subscription.add(
      "aboutItemRevisionClient",
      prom
    );

    const aboutItemRevisionModel = await promiseWrapper.promise;
    this.onFetchAboutItemSuccess(aboutItemRevisionModel);

    return aboutItemRevisionModel;
  }

  onFetchAboutItemSuccess(data: AboutItemRevisionModel) {
    this.setState({
      aboutItemRevisionModel: { kind: "success", content: data }
    });
  }

  async fetchAccResourceGroups(item: AboutItemRevisionModel) {
    const params: AccessibilityRevisionModel = {
      interactionType: item.AboutItemMetadata.interactionType,
      subject: item.AboutItemMetadata.subject,
      gradeLevel: item.AboutItemMetadata.intendedGrade,
      allowCalculator: this.makeBool(item),
      itemKey: item.AboutItemMetadata.identifier,
      bankKey: item.bankKey
    };
    const prom = this.props.accessibilityClient(params);
    const promiseWrapper = this.subscription.add("accessibilityClient", prom);
    const accessibilityResources = await promiseWrapper.promise;
    this.onFetchAccResourceSuccess(accessibilityResources);

    return accessibilityResources;
  }

  // Changes allowCalculator from "yes"/"no"/null to bool

  makeBool(item: AboutItemRevisionModel) {
    if (item.AboutItemMetadata.allowCalculator === "Yes") {
      return true;
    }

    return false;
  }

  onFetchAccResourceSuccess(data: AccResourceGroupModel[]) {
    this.setState({ accResourceGroups: { kind: "success", content: data } });
  }

  async fetchRevisions(item: ItemRevisionModel) {
    const prom = this.props.revisionsClient(item);
    const promiseWrapper = this.subscription.add("revisionsClient", prom);
    const revisions = await promiseWrapper.promise;
    this.onFetchRevisionsSuccess(revisions);

    return revisions;
  }

  onFetchRevisionsSuccess(data: RevisionModel[]) {
    this.setState({ revisions: { kind: "success", content: data } });
  }

  async fetchNamespaces() {
    const prom = this.props.namespacesClient();
    const promiseWrapper = this.subscription.add("namespacesClient", prom);
    const namespaces = await promiseWrapper.promise;
    this.onFetchNamespacesSuccess(namespaces);

    return namespaces;
  }

  onFetchNamespacesSuccess(data: NamespaceModel[]) {
    this.setState({ namespaces: { kind: "success", content: data } });
  }

  async fetchSections() {
    const prom = this.props.sectionsClient();
    const promiseWrapper = this.subscription.add("sectionsClient", prom);
    const sections = await promiseWrapper.promise;
    this.onFetchSectionsSuccess(sections);

    return sections;
  }

  onFetchSectionsSuccess(data: SectionModel[]) {
    this.setState({ sections: { kind: "success", content: data } });
  }

  onError(err: string, cb?: () => void) {
    if (err !== "Canceled") {
      this.setState(
        {
          hasError: true
        },
        cb
      );
    }
  }

  handleUpdateCsvText = (csvText: string) => {
    this.setState({ csvText });
  };

  handleUpdateItems = (items: ItemRevisionModel[]) => {
    const currentItem = items.length > 0 ? items[0] : undefined;
    const lastItem = items[items.length - 1];
    items.forEach(item => {
      if (!validItemRevisionModel(item) && item !== lastItem) {
        item.valid = false;
      } else {
        item.valid = true;
      }
    });
    if (validItemRevisionModel(lastItem) || items.length === 0) {
      items.push({});
    }
    this.setState({ items, currentItem }, () => {
      this.handleChangeViewItem();
      this.handleChangeRevision();
    });
  };

  componentWillUpdate(
    props: ItemBankContainerProps,
    state: ItemBankContainerState
  ) {
    if (
      !state.currentItem &&
      !state.nextItem &&
      !state.previousItem &&
      this.state.currentItem
    ) {
      this.props.resetUrl();
    } else if (state.items !== this.state.items && state.items.length > 1) {
      this.handleChangeViewItem();
    }
  }

  deleteItem = (key: number) => {
    this.setState(state => {
      let currentItem = state.currentItem;
      let nextItem = state.nextItem;
      let previousItem = state.previousItem;
      if (itemsAreEqual(state.items[key], currentItem)) {
        if (state.previousItem) {
          currentItem = state.previousItem;
        } else if (state.nextItem) {
          currentItem = state.nextItem;
        } else if (state.items.length === 2) {
          currentItem = undefined;
          nextItem = undefined;
          previousItem = undefined;
        }
      }

      return {
        currentItem,
        nextItem,
        previousItem,
        items: state.items.filter((i, index) => key !== index)
      };
    });
  };

  clearItems = () => {
    this.setState({
      items: [{}],
      previousItem: undefined,
      nextItem: undefined,
      currentItem: undefined
    });
  };

  /**
   * Updates prev and next items. Updates rubric, about item, and item url
   * @memberof ItemBankContainer
   */
  handleChangeViewItem = () => {
    const { currentItem, items } = this.state;
    let index = 0;

    if (currentItem) {
      this.fetchAboutItemRevisionModel(currentItem)
        .then(aboutItem => {
          this.fetchAccResourceGroups(aboutItem)
            .then(accGroups => this.handleUpdateIsaap(accGroups))
            .catch(e => this.onError(e));
          this.updateNavigationItems();
        })
        .catch(e => {
          this.onError(e, () => {
            if (items.length > 1) {
              index = items.findIndex(i => i === currentItem);
              items[index].valid = false;
              this.setState({ items }, this.updateNavigationItems);
            }
          });
        });
    }
  };

  updateNavigationItems = () => {
    const { currentItem, items } = this.state;
    if (currentItem) {
      const nextItem = getNextItemBank(currentItem, items);
      const previousItem = getPreviousItemBank(currentItem, items);

      this.setState({ currentItem, nextItem, previousItem });
    }
  };

  handleUpdateIsaap = (accGroups: AccResourceGroupModel[]) => {
    const { currentItem } = this.state;

    if (currentItem) {
      const isaap = toiSAAP(accGroups);
      this.props.setUrl(currentItem);
      this.setState({ currentItem });
      this.bubbleEventHandler(currentItem, isaap);
    }
  };

  bubbleEventHandler = (currentItem: ItemRevisionModel, isaap: string) => {
    const newItem = { ...currentItem, isaap };
    const x = document;
    const event = new CustomEvent("acc-update", { detail: newItem.isaap });
    if (x !== null) {
      x.dispatchEvent(event);
    }
  };

  handleChangeRevision = () => {
    const { currentItem } = this.state;
    if (currentItem) {
      this.fetchRevisions(currentItem).catch(e => this.onError(e));
    }
  };

  onAccessibilityUpdate = (accResourceGroups: AccResourceGroupModel[]) => {
    this.setState(
      {
        accResourceGroups: {
          kind: "success",
          content: accResourceGroups
        }
      },
      () => this.handleUpdateIsaap(accResourceGroups)
    );
  };

  onAccessibilityReset = () => {
    const {
      aboutItemRevisionModel,
      currentItem,
      accResourceGroups
    } = this.state;
    const aboutItem = getResourceContent(aboutItemRevisionModel);
    const accResources = getResourceContent(accResourceGroups);
    if (aboutItem && accResources) {
      this.setState({ currentItem: { ...currentItem, isaap: undefined } }, () =>
        this.handleUpdateIsaap(accResources)
      );
    }
  };

  handleNextItem() {
    const { items, currentItem } = this.state;
    let nextItem = currentItem;
    if (currentItem) {
      const findNext = getNextItemBank(currentItem, items);
      nextItem = findNext || nextItem;
    } else {
      nextItem = items[0];
    }

    this.setState({ currentItem: nextItem }, () => {
      this.handleChangeViewItem();
      this.handleChangeRevision();
    });
  }

  handlePreviousItem() {
    const { items, currentItem } = this.state;
    let previousItem = currentItem;
    if (currentItem) {
      const findPrevious = getPreviousItemBank(currentItem, items);
      previousItem = findPrevious || previousItem;
    } else {
      previousItem = items[0];
    }

    this.setState({ currentItem: previousItem }, () => {
      this.handleChangeViewItem();
      this.handleChangeRevision();
    });
  }

  onDirectionSelect = (direction: "next" | "previous") => {
    switch (direction) {
      case "next":
        this.handleNextItem();
        break;
      case "previous":
        this.handlePreviousItem();
        break;
      default:
        throw new Error("invalid direction");
    }
  };

  onItemSelect = (item: string) => {
    const { revisions, items } = this.state;
    let { currentItem } = this.state;
    if (currentItem) {
      currentItem = items.find(i => itemRevisionKey(i) === item);
    }
    this.setState({ currentItem }, () => {
      this.handleChangeViewItem();
      this.handleChangeRevision();
    });
  };

  onRevisionSelect = (revision: string) => {
    const { currentItem, revisions } = this.state;
    let revisionContent = getResourceContent(revisions);
    if (currentItem && revisionContent) {
      currentItem.revision = revision;
      revisionContent = revisionContent.map(r => {
        return { ...r, selected: r.commitHash === revision };
      });
    }
    this.setState(
      { currentItem, revisions: { kind: "success", content: revisionContent } },
      this.handleChangeViewItem
    );
  };

  renderItemBankEntry() {
    const { namespaces, sections, items, csvText } = this.state;
    let content: JSX.Element | undefined;

    const namespacesContent = getResourceContent(namespaces);
    const sectionsContent = getResourceContent(sections);
    if (namespacesContent && sectionsContent) {
      content = (
        <ItemBankEntry
          updateCsvText={this.handleUpdateCsvText}
          updateItems={this.handleUpdateItems}
          namespaces={namespacesContent}
          sections={sectionsContent}
          csvText={csvText}
          items={items}
          deleteItem={this.deleteItem}
          clearItems={this.clearItems}
        />
      );
    }

    return content;
  }

  renderItemBankViewer() {
    const {
      aboutItemRevisionModel,
      accResourceGroups,
      revisions,
      currentItem,
      previousItem,
      nextItem,
      items
    } = this.state;
    let content: JSX.Element | undefined;

    const aboutItemContent = getResourceContent(aboutItemRevisionModel);
    const accResourceGroupsContent = getResourceContent(accResourceGroups);
    const revisionsContent = getResourceContent(revisions);

    content = (
      <ItemBankViewer
        onAccessibilityUpdate={this.onAccessibilityUpdate}
        onAccessibilityReset={this.onAccessibilityReset}
        onDirectionSelect={this.onDirectionSelect}
        onRevisionSelect={this.onRevisionSelect}
        onItemSelect={this.onItemSelect}
        itemUrl={this.props.itemViewUrl}
        aboutItemRevisionModel={aboutItemContent}
        accResourceGroups={accResourceGroupsContent}
        revisions={revisionsContent}
        nextItem={nextItem}
        prevItem={previousItem}
        currentItem={currentItem}
        items={items}
      />
    );

    return content;
  }

  render() {
    return (
      <div className="item-container container">
        <div>
          {this.renderItemBankEntry()}
          {this.renderItemBankViewer()}
        </div>
      </div>
    );
  }
}
