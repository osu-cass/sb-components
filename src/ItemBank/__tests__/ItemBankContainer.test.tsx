import "jsdom-global/register";
import * as React from "react";
import { shallow, mount } from "enzyme";
import {
  ItemBankContainer,
  ItemBankContainerProps,
  ItemBankViewer,
  ItemBankViewerProps,
  ItemBankEntry,
  ItemRevisionModel
} from "@src/index";
import * as BankMocks from "./mocks";
import { ItemBankContainerState } from "../ItemBankContainer";
const item: ItemRevisionModel = {
  itemKey: 187,
  bankKey: 3000,
  hasBankKey: true,
  namespace: "itemreviewviewer",
  section: "math",
  revision: "asfe",
  isaap: ""
};
const item2 = { ...item, bankKey: 2332, revision: "asdf" };

describe("ItemBankContainer", () => {
  const props: ItemBankContainerProps = {
    aboutItemRevisionClient: BankMocks.mockBankAboutItemClient,
    accessibilityClient: BankMocks.mockBankAccessibilityClient,
    revisionsClient: BankMocks.mockBankRevisionsClient,
    namespacesClient: BankMocks.mockBankNamespacesClient,
    sectionsClient: BankMocks.mockBankSectionsClient,
    itemViewUrl: "hello",
    items: [item, item2],
    setUrl: BankMocks.mockOnItemHandler,
    resetUrl: () => {
      const url = "";
    }
  };

  const wrapper = mount(<ItemBankContainer {...props} />);
  it("default wrapper", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has item bank viewer", () => {
    const itemBankViewer = wrapper.findWhere(
      node => node.type() === ItemBankViewer
    );
    expect(itemBankViewer).toBeDefined();
    expect(itemBankViewer).toMatchSnapshot();
  });

  it("item bank entry", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    const itemBankEntry = wrapper.findWhere(
      node => node.type() === ItemBankEntry
    );
    expect(itemBankEntry).toBeDefined();
    expect(itemBankEntry).toMatchSnapshot();
  });

  it("updates item url", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    wrapperInstance.handleUpdateIsaap(BankMocks.mockAccessibility);
    expect(BankMocks.mockOnItemHandler).toHaveBeenCalled();
  });

  it("has current item", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    expect(wrapperInstance.state.currentItem).toBeDefined();
    expect(wrapperInstance.state.currentItem).toEqual(item);
  });

  it("updates next item", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    wrapperInstance.handleNextItem();
    expect(wrapperInstance.state.currentItem).toBeDefined();
    expect(wrapperInstance.state.currentItem).toEqual(item2);
  });

  it("item viewer calls item select next", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    const itemBankViewer = wrapper.findWhere(
      node => node.type() === ItemBankViewer
    );
    const itemBankViewerInstance = itemBankViewer.instance() as ItemBankViewer;
    itemBankViewerInstance.props.onDirectionSelect("next");
    expect(wrapperInstance.state.currentItem).toEqual(item2);
  });

  it("item viewer calls item select previous", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    wrapperInstance.handleNextItem();
    const itemBankViewer = wrapper.findWhere(
      node => node.type() === ItemBankViewer
    );
    const itemBankViewerInstance = itemBankViewer.instance() as ItemBankViewer;
    itemBankViewerInstance.props.onDirectionSelect("previous");
    expect(wrapperInstance.state.currentItem).toEqual(item);
  });

  it("item viewer calls item select revision", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    wrapperInstance.handleChangeRevision();
    const itemBankViewer = wrapper.findWhere(
      node => node.type() === ItemBankViewer
    );
    const itemBankViewerInstance = itemBankViewer.instance() as ItemBankViewer;
    itemBankViewerInstance.props.onRevisionSelect("asdf");
    item.revision = "asdf";
    expect(wrapperInstance.state.currentItem).toEqual(item);
  });

  it("item viewer calls item select previous null item", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    wrapperInstance.setState({ currentItem: undefined });
    const itemBankViewer = wrapper.findWhere(
      node => node.type() === ItemBankViewer
    );
    const itemBankViewerInstance = itemBankViewer.instance() as ItemBankViewer;
    itemBankViewerInstance.props.onDirectionSelect("previous");
    expect(wrapperInstance.state.currentItem).toEqual(item);
  });

  it("item viewer calls item select next null item", () => {
    const wrapperInstance = wrapper.instance() as ItemBankContainer;
    wrapperInstance.setState({ currentItem: undefined });
    const itemBankViewer = wrapper.findWhere(
      node => node.type() === ItemBankViewer
    );
    const itemBankViewerInstance = itemBankViewer.instance() as ItemBankViewer;
    itemBankViewerInstance.props.onDirectionSelect("next");
    expect(wrapperInstance.state.currentItem).toEqual(item);
  });
});
