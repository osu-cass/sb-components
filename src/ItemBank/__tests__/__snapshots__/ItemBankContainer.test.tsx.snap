// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemBankContainer default wrapper 1`] = `
<ItemBankContainer
  aboutItemRevisionClient={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "bankKey": 3000,
            "hasBankKey": true,
            "isaap": "",
            "itemKey": 187,
            "namespace": "itemreviewviewer",
            "revision": "asfe",
            "section": "math",
            "valid": false,
          },
        ],
      ],
    }
  }
  accessibilityClient={[MockFunction]}
  itemViewUrl="hello"
  items={
    Array [
      Object {
        "bankKey": 3000,
        "hasBankKey": true,
        "isaap": "",
        "itemKey": 187,
        "namespace": "itemreviewviewer",
        "revision": "asfe",
        "section": "math",
        "valid": false,
      },
      Object {
        "bankKey": 2332,
        "hasBankKey": true,
        "isaap": "",
        "itemKey": 187,
        "namespace": "itemreviewviewer",
        "revision": "asdf",
        "section": "math",
      },
    ]
  }
  namespacesClient={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  revisionsClient={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "bankKey": 3000,
            "hasBankKey": true,
            "isaap": "",
            "itemKey": 187,
            "namespace": "itemreviewviewer",
            "revision": "asfe",
            "section": "math",
            "valid": false,
          },
        ],
      ],
    }
  }
  sectionsClient={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  setUrl={[MockFunction]}
>
  <div
    className="item-container container"
  >
    <div>
      <ItemBankViewer
        currentItem={
          Object {
            "bankKey": 3000,
            "hasBankKey": true,
            "isaap": "",
            "itemKey": 187,
            "namespace": "itemreviewviewer",
            "revision": "asfe",
            "section": "math",
            "valid": false,
          }
        }
        itemUrl="hello"
        items={
          Array [
            Object {
              "bankKey": 3000,
              "hasBankKey": true,
              "isaap": "",
              "itemKey": 187,
              "namespace": "itemreviewviewer",
              "revision": "asfe",
              "section": "math",
              "valid": false,
            },
            Object {
              "bankKey": 2332,
              "hasBankKey": true,
              "isaap": "",
              "itemKey": 187,
              "namespace": "itemreviewviewer",
              "revision": "asdf",
              "section": "math",
            },
          ]
        }
        onAccessibilityReset={[Function]}
        onAccessibilityUpdate={[Function]}
        onDirectionSelect={[Function]}
        onItemSelect={[Function]}
        onRevisionSelect={[Function]}
      >
        <div
          className="section revisions-items"
        >
          <RevisionContainer
            onRevisionSelect={[Function]}
            revisions={Array []}
          >
            <div
              className="revisions section section-light section-compact"
            >
              <h3
                className="revisions-header"
              >
                Revisions
              </h3>
              <ul />
            </div>
          </RevisionContainer>
          <div
            aria-live="polite"
            aria-relevant="additions"
            className="section item-map-iframe"
          >
            <div
              aria-label="Toolbar with button groups"
              className="item-nav"
              role="toolbar"
            >
              <div
                aria-label="First group"
                className="item-nav-left-group"
                role="group"
              />
              <div
                className="nav-buttons"
              >
                <button
                  className="btn btn-link previous"
                  disabled={true}
                  onClick={[Function]}
                >
                  <div
                    className="nav-button-container"
                  >
                    <span
                      aria-hidden="true"
                      className="fa fa-arrow-left"
                    />
                    <div
                      className="nav-item-name"
                    >
                      Previous
                    </div>
                  </div>
                </button>
                <Component
                  className="item-dropdown"
                  label="Items"
                  labelClass="display-none"
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "disabled": true,
                        "label": "Select an Item",
                        "selected": false,
                        "value": "N/A",
                      },
                      Object {
                        "key": "3000-187-math",
                        "label": "itemreviewviewer:3000-187",
                        "selected": true,
                        "value": "3000-187-math",
                      },
                      Object {
                        "key": "2332-187-math",
                        "label": "itemreviewviewer:2332-187",
                        "selected": false,
                        "value": "2332-187-math",
                      },
                    ]
                  }
                  selected="3000-187-math"
                  wrapperClass="select-dd"
                >
                  <div
                    className="select-dd"
                  >
                    <label>
                      <span
                        className="display-none"
                      >
                        Items
                      </span>
                    </label>
                    <select
                      className="form-control item-dropdown"
                      onChange={[Function]}
                      value="3000-187-math"
                    >
                      <Component
                        disabled={true}
                        key="N/A"
                        label="Select an Item"
                        selected={false}
                        value="N/A"
                      >
                        <option
                          aria-selected={false}
                          disabled={true}
                          label="Select an Item"
                          value="N/A"
                        >
                          Select an Item
                        </option>
                      </Component>
                      <Component
                        key="3000-187-math"
                        label="itemreviewviewer:3000-187"
                        selected={true}
                        value="3000-187-math"
                      >
                        <option
                          aria-selected={true}
                          label="itemreviewviewer:3000-187"
                          value="3000-187-math"
                        >
                          itemreviewviewer:3000-187
                        </option>
                      </Component>
                      <Component
                        key="2332-187-math"
                        label="itemreviewviewer:2332-187"
                        selected={false}
                        value="2332-187-math"
                      >
                        <option
                          aria-selected={false}
                          label="itemreviewviewer:2332-187"
                          value="2332-187-math"
                        >
                          itemreviewviewer:2332-187
                        </option>
                      </Component>
                    </select>
                  </div>
                </Component>
                <button
                  className="btn btn-link next"
                  disabled={true}
                  onClick={[Function]}
                >
                  <div
                    className="nav-button-container"
                  >
                    <span
                      aria-hidden="true"
                      className="fa fa-arrow-right"
                    />
                    <div
                      className="nav-item-name"
                    >
                      Next
                    </div>
                  </div>
                </button>
              </div>
              <div
                aria-label="Second group"
                className="item-nav-right-group"
                role="group"
              />
            </div>
            <ItemViewerFrame
              title="item viewer"
              url="hello"
            >
              <div
                className="itemViewerFrame"
                tabIndex={0}
              >
                <div
                  className="loader-icon"
                >
                  <div
                    className="loader"
                  />
                </div>
                <iframe
                  className="itemviewer-iframe"
                  id="itemviewer-iframe"
                  onLoad={[Function]}
                  onLoadStart={[Function]}
                  scrolling="no"
                  src="hello"
                  title="item viewer"
                />
              </div>
            </ItemViewerFrame>
          </div>
        </div>
      </ItemBankViewer>
    </div>
  </div>
</ItemBankContainer>
`;

exports[`ItemBankContainer has item bank viewer 1`] = `
<ItemBankViewer
  currentItem={
    Object {
      "bankKey": 3000,
      "hasBankKey": true,
      "isaap": "",
      "itemKey": 187,
      "namespace": "itemreviewviewer",
      "revision": "asfe",
      "section": "math",
      "valid": false,
    }
  }
  itemUrl="hello"
  items={
    Array [
      Object {
        "bankKey": 3000,
        "hasBankKey": true,
        "isaap": "",
        "itemKey": 187,
        "namespace": "itemreviewviewer",
        "revision": "asfe",
        "section": "math",
        "valid": false,
      },
      Object {
        "bankKey": 2332,
        "hasBankKey": true,
        "isaap": "",
        "itemKey": 187,
        "namespace": "itemreviewviewer",
        "revision": "asdf",
        "section": "math",
      },
    ]
  }
  onAccessibilityReset={[Function]}
  onAccessibilityUpdate={[Function]}
  onDirectionSelect={[Function]}
  onItemSelect={[Function]}
  onRevisionSelect={[Function]}
>
  <div
    className="section revisions-items"
  >
    <RevisionContainer
      onRevisionSelect={[Function]}
      revisions={Array []}
    >
      <div
        className="revisions section section-light section-compact"
      >
        <h3
          className="revisions-header"
        >
          Revisions
        </h3>
        <ul />
      </div>
    </RevisionContainer>
    <div
      aria-live="polite"
      aria-relevant="additions"
      className="section item-map-iframe"
    >
      <div
        aria-label="Toolbar with button groups"
        className="item-nav"
        role="toolbar"
      >
        <div
          aria-label="First group"
          className="item-nav-left-group"
          role="group"
        />
        <div
          className="nav-buttons"
        >
          <button
            className="btn btn-link previous"
            disabled={true}
            onClick={[Function]}
          >
            <div
              className="nav-button-container"
            >
              <span
                aria-hidden="true"
                className="fa fa-arrow-left"
              />
              <div
                className="nav-item-name"
              >
                Previous
              </div>
            </div>
          </button>
          <Component
            className="item-dropdown"
            label="Items"
            labelClass="display-none"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "disabled": true,
                  "label": "Select an Item",
                  "selected": false,
                  "value": "N/A",
                },
                Object {
                  "key": "3000-187-math",
                  "label": "itemreviewviewer:3000-187",
                  "selected": true,
                  "value": "3000-187-math",
                },
                Object {
                  "key": "2332-187-math",
                  "label": "itemreviewviewer:2332-187",
                  "selected": false,
                  "value": "2332-187-math",
                },
              ]
            }
            selected="3000-187-math"
            wrapperClass="select-dd"
          >
            <div
              className="select-dd"
            >
              <label>
                <span
                  className="display-none"
                >
                  Items
                </span>
              </label>
              <select
                className="form-control item-dropdown"
                onChange={[Function]}
                value="3000-187-math"
              >
                <Component
                  disabled={true}
                  key="N/A"
                  label="Select an Item"
                  selected={false}
                  value="N/A"
                >
                  <option
                    aria-selected={false}
                    disabled={true}
                    label="Select an Item"
                    value="N/A"
                  >
                    Select an Item
                  </option>
                </Component>
                <Component
                  key="3000-187-math"
                  label="itemreviewviewer:3000-187"
                  selected={true}
                  value="3000-187-math"
                >
                  <option
                    aria-selected={true}
                    label="itemreviewviewer:3000-187"
                    value="3000-187-math"
                  >
                    itemreviewviewer:3000-187
                  </option>
                </Component>
                <Component
                  key="2332-187-math"
                  label="itemreviewviewer:2332-187"
                  selected={false}
                  value="2332-187-math"
                >
                  <option
                    aria-selected={false}
                    label="itemreviewviewer:2332-187"
                    value="2332-187-math"
                  >
                    itemreviewviewer:2332-187
                  </option>
                </Component>
              </select>
            </div>
          </Component>
          <button
            className="btn btn-link next"
            disabled={true}
            onClick={[Function]}
          >
            <div
              className="nav-button-container"
            >
              <span
                aria-hidden="true"
                className="fa fa-arrow-right"
              />
              <div
                className="nav-item-name"
              >
                Next
              </div>
            </div>
          </button>
        </div>
        <div
          aria-label="Second group"
          className="item-nav-right-group"
          role="group"
        />
      </div>
      <ItemViewerFrame
        title="item viewer"
        url="hello"
      >
        <div
          className="itemViewerFrame"
          tabIndex={0}
        >
          <div
            className="loader-icon"
          >
            <div
              className="loader"
            />
          </div>
          <iframe
            className="itemviewer-iframe"
            id="itemviewer-iframe"
            onLoad={[Function]}
            onLoadStart={[Function]}
            scrolling="no"
            src="hello"
            title="item viewer"
          />
        </div>
      </ItemViewerFrame>
    </div>
  </div>
</ItemBankViewer>
`;

exports[`ItemBankContainer item bank entry 1`] = `null`;
